seismic_hazard_analysis/site_source.py

```diff
@@ -0,0 +1,977 @@
+                np.dot(n, (site_nztm_coords - cur_segment_nztm_coords[0, :]))
+                / np.linalg.norm(n)
+            ) / 1e3
+            rjb_values[i] = 0
+
+        # Compute distances
+        A_ind = np.asarray([0, 0, 3, 3])
+        B_ind = np.asarray([1, 2, 1, 2])
+        rrup_values[i] = compute_min_line_segment_distance(
+            cur_segment_nztm_coords, site_nztm_coords, A_ind, B_ind
+        )
+        rjb_values[i] = compute_min_line_segment_distance(
+            cur_segment_nztm_coords[:, :2], site_nztm_coords[:2], A_ind, B_ind
+        )
+
+    return rjb_values, rrup_values
+
+
+@nb.njit(nb.float64[::1](nb.float64[::1], nb.float64[::1], nb.float64[::1]))
+def compute_segment_weight(
+    segment_ry: np.ndarray, segment_rx: np.ndarray, segment_trace_length: np.ndarray
+):
+    """
+    Computes the normalized segment weights using
+    equation 4 and 5 from Spudich et al. (2015)
+    """
+    sw = (
+        np.arctan((segment_trace_length - segment_ry) / segment_rx)
+        - np.arctan(-segment_ry / segment_rx)
+    ) / segment_rx
+
+    # Site is on the extension of the segment
+    m = np.isclose(segment_rx, 0.0)
+    if np.any(m & ((segment_ry < 0.0) | (segment_ry > segment_trace_length))):
+        sw[m] = 1 / (segment_ry[m] - segment_trace_length[m]) - 1.0 / segment_rx[m]
+
+    return sw / np.sum(sw)
+
+
+@nb.njit(
+    nb.types.UniTuple(nb.float64, 4)(
+        nb.int64[:],
+        nb.float64[:, :, :],
+        nb.float64[:, :],
+        nb.float64[:],
+        nb.int64[:],
+        nb.float64[:],
+        nb.float64[:],
+        nb.float64[:],
+        nb.float64[:],
+        nb.float64[:, :],
+    )
+)
+def compute_single_scenario_distances(
+    section_ids: np.ndarray,
+    segment_nztm_coords: np.ndarray,
+    segment_strike_vec: np.ndarray,
+    segment_trace_length: np.ndarray,
+    segment_section_ids: np.ndarray,
+    segment_rjb: np.ndarray,
+    segment_rrup: np.ndarray,
+    segment_rx: np.ndarray,
+    segment_ry: np.ndarray,
+    segment_ry_origin: np.ndarray,
+):
+    """
+    Computes the distances for the scenario defined
+    by the given section ids
+
+    Parameters
+    ----------
+    section_ids: array of ints
+        Ids of the rupture sections for this scenario
+    segment_nztm_coords: array of floats
+        All rupture segment coordinates (NZTM)
+    segment_strike_vec: array of floats
+        The strike vector for each segment
+    segment_trace_length: array of floats
+        The trace length for each segment
+    segment_section_ids: array of ints
+        The section id for each segment
+    segment_rjb: array of floats
+        Rjb distance for each segment
+    segment_rrup
+        Rrup distance for each segment
+    segment_rx
+        Rx distance for each segment
+    segment_ry
+        Ry distance for each segment
+    segment_ry_origin
+        The origin used for the
+        segment Ry calculations
+
+    Returns
+    -------
+    rjb: float
```

Comment:
in km, I believe?  It can be obvious, but there are few other places where the unit used could be documented explicitly.